distributed/core.py

```diff
@@ -109,6 +110,145 @@ def _expects_comm(func: Callable) -> bool:
+
+        Parameters
+        ----------
+        afunc
+            Coroutine function to schedule.
+        *args
+            Arguments to be passed to `afunc`.
+        **kwargs
+            Keyword arguments to be passed to `afunc`
+
+        Returns
+        -------
+            The scheduled Task object.
+
+        Raises
+        ------
+        AsyncTaskGroupClosedError
+            If the task group is closed.
+        """
+        if self.closed:  # Avoid creating a coroutine
+            raise AsyncTaskGroupClosedError(
+                "Cannot schedule a new coroutine function as the group is already closed."
+            )
+        task = self._get_loop().create_task(afunc(*args, **kwargs))
+        task.add_done_callback(self._ongoing_tasks.remove)
+        self._ongoing_tasks.add(task)
+        return task
+
+    def call_later(
+        self, delay: float, afunc: Callable[..., Coroutine], *args, **kwargs
+    ) -> asyncio.Task:
+        """Schedule a coroutine function to be executed after `delay` seconds as an `asyncio.Task`.
+
+        The coroutine function `afunc` is scheduled with `args` arguments and `kwargs` keyword arguments
+        as an `asyncio.Task` that is executed after `delay` seconds.
+
+        Parameters
+        ----------
+        delay
+            Delay in seconds.
+        afunc
+            Coroutine function to schedule.
+        *args
+            Arguments to be passed to `afunc`.
+        **kwargs
+            Keyword arguments to be passed to `afunc`
+
+        Returns
+        -------
+            The scheduled Task object.
+
+        Raises
+        ------
+        AsyncTaskGroupClosedError
+            If the task group is closed.
+        """
+        return self.call_soon(delayed(afunc, delay), *args, **kwargs)
+
+    def close(self) -> None:
+        """Closes the task group so that no new tasks can be scheduled.
+
+        Existing tasks continue to run.
+        """
+        self.closed = True
+
+    async def stop(self, timeout=1) -> None:
+        """Close the group and stop all currently running tasks.
+
+        Closes the task group and waits `timeout` seconds for all tasks to gracefully finish.
+        After the timeout, all remaining tasks are cancelled.
+        """
+        self.close()
+
+        current_task = asyncio.current_task(self._get_loop())
+        tasks_to_stop = [t for t in self._ongoing_tasks if t is not current_task]
+
+        if tasks_to_stop:
+            # Wrap gather in task to avoid Python3.8 issue,
+            # see https://github.com/dask/distributed/pull/6478#discussion_r885696827
+            async def gather():
+                return await asyncio.gather(*tasks_to_stop, return_exceptions=True)
+
+            try:
+                await asyncio.wait_for(
+                    gather(),
+                    timeout,
+                )
+            except asyncio.TimeoutError:
+                await asyncio.gather(*tasks_to_stop, return_exceptions=True)
```

Comment:
Might be good to comment that the timeout on `gather` will cancel all the tasks, so this shouldn't hang forever; that's not very intuitive.